### PR TITLE
Add gasleft to FunctionType::richIdentifier().

### PR DESCRIPTION
**Pull request by ekpyron**
Refs #3660.

This hopefully fixes the issue in #3660. This pr does not close the issue, since we may want to continue the discussion about how to avoid such problems using a test, if at all possible.

### CODE REVIEW

File: libsolidity/ast/Types.cpp [hunk 1]
```diff
@@ -2344,6 +2344,7 @@ string FunctionType::richIdentifier() const
 	case Kind::Log2: id += "log2"; break;
 	case Kind::Log3: id += "log3"; break;
 	case Kind::Log4: id += "log4"; break;
+    case Kind::GasLeft: id += "gasleft"; break;
```
Comment (chriseth) [inline libsolidity/ast/Types.cpp]: Please use tabs for indentation.

**Comment by ekpyron** [comment]
Fixed identation.

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
@ekpyron please use `Fixes #3660` next time so merging this closes the issue. (cc @bit-shift)

**Comment by ekpyron** [comment]
@axic I intentionally did not close the issue with this pull request, in order to potentially continue the discussion about a test case here... (see first comment).

**Comment by axic** [comment]
I am sorry, didn't read that.

**Pull request merged**